apps/keyboard/test/unit/imes/latin/worker_test.js

```diff
@@ -451,22 +448,13 @@ suite('Latin worker', function() {
       callback = filter;
       filter = undefined;
     }
-    var occ = worker.onmessage.callCount;
-    var iv = setInterval(function() {
-      if (worker.onmessage.callCount > occ) {
-        var ev = worker.onmessage.args[worker.onmessage.callCount - 1][0];
-        // console.log(ev.data);
-
-        occ = worker.onmessage.callCount;
-
-        if (filter !== undefined && ev.data.cmd !== filter) {
-          return;
-        }
-
-        clearInterval(iv);
-        callback(ev.data);
+    worker.onmessage = function handleWorkerMessage(evt) {
```

Comment:
what do you think of using `addEventListener('message', ...)` instead of `onmessage` ?